spec/rules_spec.rb

```diff
@@ -0,0 +1,146 @@
+      board = Board.new
+      board.mark(3, 'X')
+      board.mark(4, 'X')
+      board.mark(5, 'X')
+      rules = Rules.new(board)
+
+      horizontal = rules.horizontal_win?
+
+      expect(horizontal).to eq(true)
+    end
+
+    it 'should end the game with a wins horizontally on the third row' do
+      board = Board.new
+      board.mark(6, 'X')
+      board.mark(7, 'X')
+      board.mark(8, 'X')
+      rules = Rules.new(board)
+
+      horizontal = rules.horizontal_win?
+
+      expect(horizontal).to eq(true)
+    end
+
+    it 'should end the game with a wins vertical on the left column' do
+      board = Board.new
+      board.mark(0, 'X')
+      board.mark(3, 'X')
+      board.mark(6, 'X')
+      rules = Rules.new(board)
+
+      vertical = rules.vertical_win?
+
+      expect(vertical).to eq(true)
+    end
+
+    it 'should end the game with a wins vertical on the middle column' do
+      board = Board.new
+      board.mark(1, 'X')
+      board.mark(7, 'X')
+      board.mark(4, 'X')
+      rules = Rules.new(board)
+
+      vertical = rules.vertical_win?
+
+      expect(vertical).to eq(true)
+    end
+
+    it 'should end the game with a wins vertical on the right colum' do
+      board = Board.new
+      board.mark(2, 'X')
+      board.mark(5, 'X')
+      board.mark(8, 'X')
+      rules = Rules.new(board)
+
+      vertical = rules.vertical_win?
+
+      expect(vertical).to eq(true)
+    end
+
+    it 'should end the game with a wins vertical on the left diagonal' do
+      board = Board.new
+      board.mark(0, 'X')
+      board.mark(4, 'X')
+      board.mark(8, 'X')
+      rules = Rules.new(board)
+
+      diagonal = rules.diagonal_win?
+
+      expect(diagonal).to eq(true)
+    end
+
+    it 'should end the game with a wins vertical on the right diagonal' do
+      board = Board.new
+      board.mark(2, 'X')
+      board.mark(4, 'X')
+      board.mark(6, 'X')
+      rules = Rules.new(board)
+
+      diagonal = rules.diagonal_win?
+
+      expect(diagonal).to eq(true)
+    end
+
+    it 'should should output true if a horizontal game is won' do
+      board = Board.new
+      board.mark(6, 'X')
+      board.mark(7, 'X')
+      board.mark(8, 'X')
+      rules = Rules.new(board)
+
+      win = rules.win?
+
+      expect(win).to eq(true)
+    end
+
+    it 'should should output true if a vertical game is won' do
+      board = Board.new
+      board.mark(0, 'X')
+      board.mark(3, 'X')
+      board.mark(6, 'X')
+      rules = Rules.new(board)
+
+      win = rules.win?
+
+      expect(win).to eq(true)
+    end
+
+    it 'should output true and game ends in a tie' do
+      board = Board.new
+      board.mark(0, 'X')
+      board.mark(1, 'X')
+      board.mark(2, 'X')
+      board.mark(3, 'X')
+      board.mark(4, 'X')
+      board.mark(5, 'X`')
+      board.mark(6, 'X')
+      board.mark(7, 'X')
+      board.mark(8, 'X')
+      rules = Rules.new(board)
+
+      win = rules.win?
```

Comment:
1. Relates to other comment above, this test should be testing "tie_end?" instead of win?
2. Also, for clarity I would replace the X with apostrophes (X') with  O's for better readability. (unless this was a typo)
3. If the apostrophe was not a typo, then the way this test is set up, lines 130-138 are all winning combination so this is always going to return true and not the best way to test for a tie

Reply:
this test was added and not finished tie has now been tested for